postgis/lwgeom_geos.c

```diff
@@ -1014,16 +1014,27 @@ Datum ST_GeneratePoints(PG_FUNCTION_ARGS)
 	LWGEOM *lwgeom_input;
 	LWGEOM *lwgeom_result;
 	int32 npoints;
+	int32 seed = 0;
 
 	gser_input = PG_GETARG_GSERIALIZED_P(0);
-	npoints = DatumGetInt32(DirectFunctionCall1(numeric_int4, PG_GETARG_DATUM(1)));
+	npoints = PG_GETARG_INT32(1);
 
 	if (npoints < 0)
 		PG_RETURN_NULL();
 
+	if (PG_NARGS() > 2 && ! PG_ARGISNULL(2))
+	{
+		seed = PG_GETARG_INT32(2);
+		if (seed < 1)
+		{
+			lwpgerror("ST_GeneratePoints: seed must greater than zero");
```

Comment:
Check is <1, message is about >0. Also, "seed must be"

Reply:
I don't see any issue with the logic, but I'll fix the wording. Basically with this implementation, a user-supplied seed value must be >0.